### PR TITLE
Fix cornercase of average precision

### LINKED ISSUE #2494: Behaviors of AUROC and Average Precision are inconsistent when all labels are equal
## 🐛 Bug

When all labels are equal (either all zeros or all ones), the current implementation of `AUROC` and `AveragePrecision` have pretty different behaviors.
When labels are all ones, AUROC gives 0, while AP gives 1.
When labels are all zeros, AUROC gives 0, while Average Precision gives NaN.

I think it is better to add a flag such that both metrics would return NaN when all labels are equal to better inform users.

### To Reproduce

```
>>> from torchmetrics import AUROC, AveragePrecision
>>> import torch
>>> auroc = AUROC(task = "binary")
>>> ap = AveragePrecision(task = "binary")
>>> preds = torch.randn(10)
>>> labels = torch.ones(10, dtype = torch.long)
>>> auroc(preds, labels)
/opt/homebrew/anaconda3/lib/python3.9/site-packages/torchmetrics/utilities/prints.py:43: UserWarning: No negative samples in targets, false positive value should be meaningless. Returning zero tensor in false positive score
  warnings.warn(*args, **kwargs)  # noqa: B028
tensor(0.)
>>> ap(preds, labels)
tensor(1.)
>>> labels = torch.zeros(10, dtype = torch.long)
>>> auroc(preds, labels)
/opt/homebrew/anaconda3/lib/python3.9/site-packages/torchmetrics/utilities/prints.py:43: UserWarning: No positive samples in targets, true positive value should be meaningless. Returning zero tensor in true positive score
  warnings.warn(*args, **kwargs)  # noqa: B028
tensor(0.)
>>> ap(preds, labels)
tensor(nan)
```

### Expected behavior

when labels are all equal, it should return a NaN all the time.
at least, there can be a flag like `equal_label_mode`.
```python
>>> ap = AveragePrecision(task = "binary", equal_label_mode = "nan")
>>> ap = AveragePrecision(task = "binary", equal_label_mode = "nan")
```
that give the expected behavior.

### Environment

- TorchMetrics version: installed via pip. '1.3.0.post0'
- Python & PyTorch Version (e.g., 1.0): Python 3.9.12,  torch 2.1.0
- Any other relevant information such as OS (e.g., Linux): Linus

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #2494
Make sure that our implementation matches sklearn implementation. 

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2507.org.readthedocs.build/en/2507/

### CODE REVIEW

File: src/torchmetrics/functional/classification/precision_recall_curve.py [hunk 1]
```diff
@@ -261,6 +262,8 @@ def _binary_precision_recall_curve_compute(
     original input, then we dynamically compute the binary classification curve.
 
     """
+    # import pdb
+    # pdb.set_trace()
```
Comment (stancld) [inline src/torchmetrics/functional/classification/precision_recall_curve.py:266]:
```suggestion
```

**Review by stancld — APPROVED** [review]

**Review by Borda — APPROVED** [review]
